src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -134,21 +134,11 @@ private void Initialize()
         {
             if (string.IsNullOrWhiteSpace(Project))
             {
-                string directory = Directory.GetCurrentDirectory();
-                string[] projectFiles = Directory.GetFiles(directory, "*.*proj");
-
-                if (projectFiles.Length == 0)
-                {
-                    var project = "--project";
-
-                    throw new GracefulException(LocalizableStrings.RunCommandExceptionNoProjects, directory, project);
-                }
-                else if (projectFiles.Length > 1)
-                {
-                    throw new GracefulException(LocalizableStrings.RunCommandExceptionMultipleProjects, directory);
-                }
-
-                Project = projectFiles[0];
+                Project = FindSingleProjectInDirectory(Directory.GetCurrentDirectory());
```

Comment:
Maybe I am being pick here, but at this point, why not do Project = Directory.GetCurrentDirectory();

And then remove the else from below and just make it an independent if? That would remove this duplicated call to FindSingleProjectInDirectory, which still feels wrong to me.

So, like this:

```
if (string.IsNullOrWhiteSpace(Project))
{
  Project = Directory.GetCurrentDirectory();
}

if (Directory.Exists(Project)) 
{
  Project = FindSingleProjectInDirectory(Project);
}